packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp

```diff
@@ -1597,6 +1598,9 @@ class NumericToolsLevelSet : public NumericToolsBase<ValueType, DeviceType> {
             const auto h_buf_factor_ptr = Kokkos::subview(_h_buf_factor_ptr, range_buf_factor_ptr);
             factorizeCholeskyOnDevice(pbeg, pend, h_buf_factor_ptr, work);
             Kokkos::fence();
+            if (rval != 0) {
+              TACHO_TEST_FOR_EXCEPTION(rval, std::runtime_error, "POTRF (team) returns non-zero error code.");
```

Comment:
Now I think that I understand the `*info = 1+p;`. Basically, if one of the pivots is wrong, the code throws an error. 

If this is correct, then if `*info!=0` the algorithm could stop (it might require an extra barrier if there is none).
Or, instead of a scalar `*info` value, you could consider a masked value (a 1D view of boolean values for instance) that would store for instance a 1 if the current problem was correctly solved and a 0 otherwise. Then, even if one problem failed, you could still access the solutions of the passed problems by using the mask.

Reply:
Thank you again, @kliegeois!! I welcome any suggestion, and will make any code change.

Just for the discussion, this is part of a larger sparse factorization code, so if one block fails, then the entire factorization is bad, at least for now.. Later on, we may consider a recovery scheme to fix the failed block factorizations. In that case, an array would give us more info, but again, I was worried about the performance overhead of checking them all.

Reply:
Thanks for the comments, a reduced OR logic as you did is enough then.
In KK batched sparse solvers we use a view but I don't think that there is a significant overhead associate to it.